lib/bitcoin_ticker.rb

```diff
@@ -6,14 +6,33 @@
 
 module BitcoinTicker
   class BitcoinTicker
+    include PriceComparer
+    DEFAULT_PRICE_THRESHOLD = 10.0
+
+    # call-seq:
+    #   BitcoinTicker.new price_threshold, slack_webhook_url
+    #
+    # Parameters:
+    #   * price_threshold   [Float] : Minimum variation of the BTC price before considering communicating it (default: 10.0)
+    #   * slack_webhook_url [String]: URL of the Slack Webhook
+    def initialize(price_threshold: DEFAULT_PRICE_THRESHOLD, slack_webhook_url:)
+      self.price_threshold   = price_threshold
+      self.notifier          = SlackNotifier.new(webhook_url: slack_webhook_url)
+      self.price_saver       = PriceSaver.new
+      self.price_checker     = PriceChecker.new
+    end
+
     def tick
-      current_price = PriceChecker.new.current_price
-      price_saver = PriceSaver.new
+      current_price = price_checker.current_price
       previous_price = price_saver.read
-      if PriceComparer.new.compare_to_threshold previous_price, current_price, ENV["BITCOIN_PRICE_THRESHOLD"].to_f
-        SlackNotifier.new.notify ENV["SLACK_WEBHOOK_URL"], current_price, current_price > previous_price
+
+      if compare_to_threshold previous_price, current_price, price_threshold
```

Comment:
If i'm inheriting from this class I wanna be able to override the comparison logic and the only way I have is to override... a private method. So this is how I'd do it I think. I'd be glad to hear your thoughts :smile: